docs/ref/request-response.txt

```diff
@@ -425,21 +425,32 @@ Methods
     Returns ``True`` if the request is secure; that is, if it was made with
     HTTPS.
 
-.. method:: HttpRequest.accepts(mime_type)
+.. method:: HttpRequest.get_preferred_type(media_types)
 
-    Returns ``True`` if the request ``Accept`` header matches the ``mime_type``
-    argument:
+    .. versionadded:: 5.2
+
+    Returns the preferred mime type from ``media_types``, based on the
+    ``Accept`` header, or ``None`` if the client does not accept any of the
+    provided types.
+
+    Assuming the client sends an ``Accept`` header of
+    ``text/html,application/json;q=0.8``:
 
     .. code-block:: pycon
 
-        >>> request.accepts("text/html")
-        True
+        >>> request.get_preferred_type(["text/html", "application/json"])
+        "text/html"
+        >>> request.get_preferred_type(["application/json", "text/plain"])
+        "application/json"
+        >>> request.get_preferred_type(["application/xml", "text/plain"])
+        None
 
     Most browsers send ``Accept: */*`` by default, so this would return
     ``True`` for all content types. Setting an explicit ``Accept`` header in
```

Comment:
I still think this is mixing up elements from `HttpRequest.accepts(mime_type)`, so example this says:

```Most browsers send ``Accept: */*`` by default, so this would return ``True`` for all content types.```

But `accepts` would return `True`, this would return the first item of `mime_types`

I think this sentence should be adapted for `HttpRequest.get_preferred_type(media_types)` and this sentence also added back in to `HttpRequest.accepts(mime_type)`

I'm wondering for we should best show that the ```If a response varies depending on the content of the ``Accept`` header...``` paragraph applies to both of these and is more general advice - maybe we should make this a note or admonition or something :thinking:

Reply:
I've tweaked the wording a little, and extracted the caching point into a `note`. Because each attribute is just listed, without grouping, it's hard to find a common place, but at least putting in on the first (and preferred) API should make it easier to see.